java/src/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/ClasspathParser.java

```diff
@@ -63,6 +64,10 @@ public Set<String> getUsedTypes() {
     return usedTypes;
   }
 
+  public Set<String> getUsedPackagesWithoutSpecificTypes() {
+    return usedPackagesWithoutSpecificTypes;
```

Comment:
For future reference, because you're returning the actual field and that's marked as being immutable, anyone calling this method can now add or remove items from the returned value. A defensive copy to `ImmutableSet` (which preserves insertion order) is normally a Good Idea.

Reply:
Good call - added the defensive copies (and changed the signatures to return `Immutable*`, which I know isn't strictly necessary, but I generally find nice to do :))